pbj-integration-tests/src/main/java/com/hedera/pbj/integration/fuzz/ObjectBuilder.java

```diff
@@ -0,0 +1,188 @@
+            if (constructors.length > 1) {
+                throw new FuzzTestException("Multiple constructors are not supported for "
+                        + clz.getName() + " : " + constructors);
+            }
+
+            Constructor<T> ctor = (Constructor<T>) constructors[0];
+            Class<?>[] parameterTypes = ctor.getParameterTypes();
+            if (parameterTypes.length == 0) {
+                throw new FuzzTestException("The model class doesn't have any fields: " + clz.getName());
+            }
+
+            Type[] genericTypes = ctor.getGenericParameterTypes();
+
+            // For OneOf types, we'll branch the list of arguments in place
+            List<List> arguments = new ArrayList<>();
+            arguments.add(new ArrayList<>());
+
+            for (int i = 0; i < parameterTypes.length; i++) {
+                BranchingValue values = BranchingValue.from(parameterTypes[i], genericTypes[i], clz);
+
+                if (values.values().size() == 1) {
+                    for (int j = 0; j < arguments.size(); j++) {
+                        arguments.get(j).add(values.values().get(0));
+                    }
+                } else {
+                    // branch "in place" by building a new arguments list...
+                    List<List> newArguments = new ArrayList<>();
+                    for (int j = 0; j < arguments.size(); j++) {
+                        List list = arguments.get(j);
+
+                        for (int k = 0; k < values.values().size(); k++) {
+                            List newList = new ArrayList(list);
+                            newList.add(values.values().get(k));
+                            newArguments.add(newList);
+                        }
+                    }
+
+                    // ...and then replacing it:
+                    arguments = newArguments;
+                }
+            }
+
+            // Finally, we've built lists of arguments lists for every model variation
+            // based on its OneOf fields. Now construct a new model object for each
+            // arguments' list:
+            List<T> objects = new ArrayList<>();
+            for (int j = 0; j < arguments.size(); j++) {
+                objects.add(ctor.newInstance(arguments.get(j).toArray()));
+            }
+
+            return objects;
+        } catch (InstantiationException
+                 | IllegalAccessException
+                 | InvocationTargetException
+                 | NoSuchMethodException e
+        ) {
+            throw new FuzzTestException("Java Reflection failed", e);
+        }
+    }
+
+    /**
+     * A helper class that represents one or several possible values
+     * for a given field type.
+     */
+    private static record BranchingValue(List<?> values) {
+
+        static BranchingValue from(Class<?> argClz, Type argType, Class<?> objClz)
+                throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
```

Comment:
I assume that the number of supported types will grow over time and this method will grow with it. 
Could you please extract creation of a `BranchingValue` for each type to a separate method? 
Like `createBytesBranchingValue`, `createOneOfBranchingValue` and `createLongBranchingValue` (this one is least necessary but should be there for the sake of consistency.

Reply:
This code is now gone.